pkg/mvtxdaemon/server/service/mvtx_service.go

```diff
@@ -33,19 +33,32 @@ import (
 	"github.com/numaproj/numaflow/pkg/apis/proto/mvtxdaemon"
 	"github.com/numaproj/numaflow/pkg/metrics"
 	raterPkg "github.com/numaproj/numaflow/pkg/mvtxdaemon/server/service/rater"
+	//runtimePkg "github.com/numaproj/numaflow/pkg/mvtxdaemon/server/service/runtime"
 	"github.com/numaproj/numaflow/pkg/shared/logging"
 )
 
 // MonoVtxPendingMetric is the metric emitted from the MonoVtx lag reader for pending stats
 // Note: Please keep consistent with the definitions in rust/monovertex/sc/metrics.rs
 const MonoVtxPendingMetric = "monovtx_pending"
 
+type PodReplica string
+
+type ErrorDetails struct {
+	Container string
+	Timestamp string
+	Code      string
+	Message   string
+	Details   string
+}
+
 type MonoVertexService struct {
 	mvtxdaemon.UnimplementedMonoVertexDaemonServiceServer
 	monoVtx       *v1alpha1.MonoVertex
 	httpClient    *http.Client
 	rater         raterPkg.MonoVtxRatable
 	healthChecker *HealthChecker
+	localCache    map[PodReplica][]ErrorDetails
```

Comment:
Thread safe?

Reply:
this will be moved to runtime service as part of this [PR](https://github.com/numaproj/numaflow/pull/2469). wil refactor once that is merged.